src/imu/sensors/accelerometer.cpp

```diff
@@ -0,0 +1,45 @@
+#include "accelerometer.hpp"
+
+namespace imu {
+
+auto Accelerometer::Init(const std::uint8_t i2c_address) noexcept -> types::DriverStatus {
+  SENSOR_DATA_REGISTER = imu::ACCEL_MEASUREMENT_DATA;
+  REGISTER_DATA_LENGTH_IN_BYTES = 6;
+  CONFIG_REGISTER = imu::ACCEL_CONFIG;
+
+  return SensorWithSensitivity::Init(i2c_address);
+}
+
+auto Accelerometer::Update(void) noexcept -> types::DriverStatus {
+  if (SensorVector::Update() == types::DriverStatus::OK) {
+    auto adc_2_accel = GetFactorADC2Accelerometer();
+    sensor_values_.x = static_cast<std::int16_t>(adc_2_accel * (float)sensor_values_.x);
+    sensor_values_.y = static_cast<std::int16_t>(adc_2_accel * (float)sensor_values_.y);
+    sensor_values_.z = static_cast<std::int16_t>(adc_2_accel * (float)sensor_values_.z);
+    return types::DriverStatus::OK;
+  } else {
+    return types::DriverStatus::HAL_ERROR;
+  }
+}
+
+auto Accelerometer::GetFactorADC2Accelerometer(void) noexcept -> float {
+  auto accel_resolution = 0.0f;
+
+  switch (sensitivity_) {
+    case types::ImuSensitivity::FINEST:
+      accel_resolution = 16.0;
```

Comment:
iinstead of having the sensitivites as magic numbers, xou could set these values to the types::ImuSensitivity enum members

Reply:
Principally correct, but I use  types::ImuSensitivity for gyroscope and accelerometer and the values are completely different. I would have to split up the types. I could add a special enum for this, but would that be better readable? The switch case would be more or less the same.

Reply:
Ah ok I see. Then the switch case would be indeed the same and readability would not be improved